### ISSUE TITLE
supports_star_without_alias issue for 

### ISSUE THREAD

**Issue by namarkus** [tidyverse/dbplyr#1003]
As described in https://github.com/tidyverse/dbplyr/issues/908 for Oracle and https://github.com/tidyverse/dbplyr/issues/966 for Teradata, specifying new columns after SELECT * is not possible with [Exasol-Database](https://github.com/exasol) neither. Most likely there are other RDBMS that do not support this.

I like the use of star instead of repeating every columnname, it's very convenient and makes it easier to inspect the generated SQL-query.  IMHO adding the alias as a prefix to the star would be a more generic solution to these problems:

Following dplyr-pipe
```
tbl(db, "my_table") %>%
  mutate(
    new_col = "any_value"
  ) %>% show_query()

```
sould translate to:
```
<SQL>
SELECT "my_table".*, 'any_value' AS new_col
FROM "my_table"
```
Of course, every derived table e.g.  `select "sub1".* from (select * from "my_table") "sub1"` has to be declared with an alias too.
Would it be possible to prefix every star with an table alias?

Thanks!

**Comment by hadley** [comment]
@namarkus are you saying that you can do `SELECT table.*, x, y, z` in Oracle and Teradata?

**Comment by namarkus** [comment]
No, that's not what I meant. The wording was a little imprecise, I'm sorry.
The prefix for the *-notation should be the table _alias_ not the table _name_. I have adjusted the name of the issue and also the example.

**Comment by namarkus** [comment]
Yes, definitely. I looked this up in the documentation for 3 different RDBMS:
**Oracle**
[https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/SELECT.html#GUID-CFA006CA-6FF1-4972-821E-6996142A51C6](url)
In the section "select_list::=" the following valid possibilities are listed:
select schema.table.* , ... 
select t_alias.* , ...

**Teradata**
[https://docs.teradata.com/r/Teradata-VantageTM-SQL-Data-Manipulation-Language/March-2019/Select-Statements/SELECT]()
According to section "B 
SELECT table_name.* , ... 
is a valid expression. Unfortunately, it is not quite clear whether table_name means the actual name of the table or an alias.

**Exasol**
[https://docs.exasol.com/db/latest/sql/select.htm]()
Section "select_list::=" names the following valid expressions:
SELECT table.* , ...
SELECT view.* , ....
(however in my Exasol system only SELECT table_alias.* works...)

In my experience (and contrary to the documentation)
SELECT table_alias.* ,...
actually works in most systems.
What is clear in all 3 syntax definitions, however, is that
SELECT *, ...
is not a valid statement.

**Comment by hadley** [comment]
@mgirlich do you think we could make this an option or maybe even the default?

**Comment by hadley** [comment]
> 4) The <select sublist> "<qualifier>.*" for some <qualifier> Q is
>    equivalent to a <value expression> sequence in which each <value
>    expression> is a <column reference> CR that references a column
>    of T that is not a common column of a <joined table>. Each col-
>    umn of T that is not a common column of a <joined table> shall
>    be referenced exactly once. The columns shall be referenced in
>    the ascending sequence of their ordinal positions within T.

Which I think implies it's valid SQL 92?

**Comment by hadley** [comment]
I think that does imply that `< qualifier > < period > < asterisk >` is ok?

**Comment by namarkus** [comment]
Yes, I agree with that.

**Comment by But2ene** [comment]
> I mean are you implying that adding the prefix fixes the problem with using a bare *?

In Teradata, yes it does.

> I think that does imply that `< qualifier > < period > < asterisk >` is ok?

Yes it works in Teradata, both table name and alias work. Qualifier/alias would be the best default for Teradata
